src/Common/src/Common.Security/CertificateConfigurationExtensions.cs

```diff
@@ -0,0 +1,109 @@
+        {
+            keys[$"{keyPrefix}PrivateKeyFilePath"] = privateKeyFilePath;
+        }
+
+        builder.AddInMemoryCollection(keys);
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds PEM files representing application identity to application configuration. When running outside of Cloud Foundry based platforms, will create
+    /// certificates resembling those found on the platform.
+    /// </summary>
+    /// <param name="builder">
+    /// Your <see cref="IConfigurationBuilder" />.
+    /// </param>
+    /// <remarks>
+    /// Outside of Cloud Foundry, CA and Intermediate certificates will be created a directory above the current project so that they can be shared between
+    /// different projects operating in the same solution context.
+    /// </remarks>
+    public static IConfigurationBuilder AddAppInstanceIdentityCertificate(this IConfigurationBuilder builder)
+    {
+        return builder.AddAppInstanceIdentityCertificate(null, null);
+    }
+
+    /// <summary>
+    /// Adds PEM files representing application identity to application configuration. When running outside of Cloud Foundry based platforms, will create
+    /// certificates resembling those found on the platform.
+    /// </summary>
+    /// <param name="builder">
+    /// Your <see cref="IConfigurationBuilder" />.
+    /// </param>
+    /// <param name="organizationId">
+    /// (Optional) A GUID representing an organization, for use with Cloud Foundry certificate-based authorization policy.
+    /// </param>
+    /// <param name="spaceId">
+    /// (Optional) A GUID representing a space, for use with Cloud Foundry certificate-based authorization policy.
+    /// </param>
+    /// <remarks>
+    /// Outside of Cloud Foundry, CA and Intermediate certificates will be created a directory above the current project so that they can be shared between
+    /// different projects operating in the same solution context.
+    /// </remarks>
+    public static IConfigurationBuilder AddAppInstanceIdentityCertificate(this IConfigurationBuilder builder, Guid? organizationId, Guid? spaceId)
+    {
+        if (!Platform.IsCloudFoundry)
+        {
+            organizationId ??= Guid.NewGuid();
+            spaceId ??= Guid.NewGuid();
+
+            var task = new LocalCertificateWriter();
+            task.Write((Guid)organizationId, (Guid)spaceId);
+
+            Environment.SetEnvironmentVariable("CF_SYSTEM_CERT_PATH",
+                Path.Combine(Directory.GetParent(LocalCertificateWriter.AppBasePath)!.ToString(), "GeneratedCertificates"));
+
+            Environment.SetEnvironmentVariable("CF_INSTANCE_CERT",
+                Path.Combine(LocalCertificateWriter.AppBasePath, "GeneratedCertificates", "SteeltoeInstanceCert.pem"));
+
+            Environment.SetEnvironmentVariable("CF_INSTANCE_KEY",
+                Path.Combine(LocalCertificateWriter.AppBasePath, "GeneratedCertificates", "SteeltoeInstanceKey.pem"));
+        }
```

Comment:
I suppose. Doesn't really matter to me either way, but it might be better to keep Steeltoe in the name so it's clear what put these files there and avoid any mysteries